k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -3616,10 +3623,14 @@ public void run() {
                         accounts = prefs.getAvailableAccounts();
                     }
 
+                    prefs.loadKeywords();
+
                     for (final Account account : accounts) {
                         checkMailForAccount(context, account, ignoreLastCheckedTime, listener);
                     }
 
+                    prefs.saveKeywords();
```

Comment:
What are we saving here?

Reply:
Thanks for spotting this! It was a leftover from the time when new keywords found on the server were initially visible. The same applies to `saveKeywords()` in `MessageListAdapter.bindView()`.

The idea was to make sure that all possibly newly added keywords are better written to the preferences, so that their otherwise undefined color and position are fixed across restarts. However, since new keywords found on the server are now initially invisible, we only need to save them just before they are shown first. This now happens in the KeywordEditor. (I hope I didn't miss anything important...)